### DISCUSSION TITLE
Local vs Global domain cell indices

### DISCUSSION THREAD

**Discussion by rajkjoshi7** [PrincetonUniversity/athena#619] [Q&A] [answered]
Hi everyone, 
I am trying to set initial conditions using a data file. I have adjusted the data file in a manner that the number of cells and domain covered by ATHENA++ is exactly similar to the grid in the data file so that I can directly import data by matching the cell indices. In the serial runs, it is not a problem, but in parallel mode, the ATHENA++ domain is divided into mesh blocks so the indices in each mesh block will not correspond to the index of the global grid. How do I translate the indices of the local mesh block to cell indices corresponding to the original (global) domain?

**Comment by tomidakn** [comment, marked as answer]
You can use MeshBlock's LogicalLocation (pmb->loc), which contains (lx1, lx2, lx3, level). If lx1 is N, it means the MeshBlock is N-th block in the x1 direction on that level. So, if your MeshBlock size is (nbx1, nbx2, nbx3), (i,j,k) cell's global index on its level can be calculated as (lx1\*nbx1+i, lx2\*nbx2+j, lx3\*nbx3+k). Note that you may need to adjust the indices according to ghost cells.
